src/Commands.pm

```diff
@@ -6564,113 +6546,144 @@ sub cmdMail {
 		return;
 	}
 
-	my ($cmd, $args_string) = @_;
-	my @args = parseArgs($args_string, 4);
-
-	# mail send
-	if ($cmd eq 'ms') {
-		unless ($args[0] && $args[1] && $args[2]) {
-			message T("Usage: ms <receiver> <title> <message>\n"), "info";
-		} else {
-			my ($receiver, $title, $msg) = ($args[0], $args[1], $args[2]);
-			$messageSender->sendMailSend($receiver, $title, $msg);
-		}
-
-	# mail open
-	} elsif ($cmd eq 'mo') {
+	my (undef, $args_string) = @_;
+	my @args = parseArgs($args_string, 3);
 
-		unless ($args[0] =~ /^\d+$/) {
-			message T("Usage: mo <mail #>\n"), "info";
-		} elsif (!$mailList->[$args[0]]) {
-			if (@{$mailList}) {
-				message TF("No mail found with index: %s. (might need to re-open mailbox)\n", $args[0]), "info";
-			} else {
-				message T("Mailbox has not been opened or is empty.\n"), "info";
-			}
+	if ($args[0] eq 'open') {
+		if (defined $mailList) {
+			error T("Your Mailbox is already opened.\n");
 		} else {
-			$messageSender->sendMailRead($mailList->[$args[0]]->{mailID});
+			message T("Sending request to open Mailbox.\n");
+			$messageSender->sendMailboxOpen();
 		}
 
-	# mail inbox => set on begin as standard?
-	} elsif ($cmd eq 'mi') {
-		# if mail not already opened needed?
+	} elsif ($args[0] eq 'refresh') {
 		$messageSender->sendMailboxOpen();
 
-	# mail window (almost useless?)
-	} elsif ($cmd eq 'mw') {
-		unless (defined $args[0]) {
-			message T("Usage: mw [0|1|2] (0:write, 1:take item back, 2:zeny input ok)\n"), "info";
-		} elsif ($args[0] =~ /^[0-2]$/) {
-			$messageSender->sendMailOperateWindow($args[0]);
+	} elsif ($args[0] eq 'read') {
+		unless ($args[1] =~ /^\d+$/) {
+			error T("Syntax Error in function 'mail read' (Mailbox)\n" .
+				"Usage: mail read <mail #>\n");
+		} elsif (!defined $mailList) {
+			warning T("Your Mailbox is not open. Use the command 'mail open'.\n");
+		} elsif (!$mailList->[$args[1]]) {
+				warning TF("No mail found with index: %s\n", $args[1]);
+		} elsif ($mailList->[$args[1]]->{mailID}) {
+			$messageSender->sendMailRead($mailList->[$args[1]]->{mailID});
+		}
+
+	} elsif ($args[0] eq 'get') {
+		unless ($args[1] =~ /^\d+$/) {
+			error T("Syntax Error in function 'mail get' (Mailbox)\n" .
+				"Usage: mail get <mail #>\n");
+		} elsif (!defined $mailList) {
+			warning T("Your Mailbox is not open. Use the command 'mail open'.\n");
+		} elsif (!$mailList->[$args[1]]) {
+				warning TF("No mail found with index: %s\n", $args[1]);
+		} elsif ($mailList->[$args[1]]->{mailID}) {
+			$messageSender->sendMailGetAttach($mailList->[$args[1]]->{mailID});
+		}
+
+	} elsif ($args[0] eq 'setzeny') {
+		if ($args[1] =~ /^\d+$/) {
```

Comment:
i think we can add a check like rodex:
https://github.com/OpenKore/openkore/blob/001a5ee9c34e2829d650ae590d1f9e60bd45a9a5/src/Commands.pm#L7344-L7347

Reply:
it's not obligatory
```
s
------------------------ Status ------------------------
merchhh                          HP: 262447/262447 (100%
Merchant (Boy)                   SP:  31606/31606 (100%)
Base: 101                         0/4,032,062 /0 (0.00%)
Job : 1                                  0/30 /0 (0.00%)
Zeny: 3,208                  Weight: 110.2/304740 (0.0%)
Statuses: On Push Cart
Spirits/Coins/Amulets: 0

Total Damage: 0              Dmg/sec: 0.00
Total Time spent (sec):      0.00
Last Monster took (sec):     0.00
--------------------------------------------------------
Done talking with .

mail open
Sending request to open Mailbox.
There is no mail in your inbox.

mail setzeny 4000
Succeeded to attach zeny.
You lost 3,208 zeny.
```